front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx

```diff
@@ -1,276 +1,278 @@
-//   const [extViewport, setExtViewport] = useState<Viewport | undefined>(undefined);
-//   const [showWarpedMap, setShowWarpedMap] = useState(false);
+  const [heightOfSpaceTimeChart, setHeightOfSpaceTimeChart] = useState(600);
+  const [heightOfSpeedSpaceChart, setHeightOfSpeedSpaceChart] = useState(250);
+  const [heightOfSimulationMap] = useState(MAP_MIN_HEIGHT);
+  const [heightOfSpaceCurvesSlopesChart, setHeightOfSpaceCurvesSlopesChart] = useState(150);
+  const [initialHeightOfSpaceCurvesSlopesChart, setInitialHeightOfSpaceCurvesSlopesChart] =
+    useState(heightOfSpaceCurvesSlopesChart);
 
-//   const [heightOfSpaceTimeChart, setHeightOfSpaceTimeChart] = useState(600);
-//   const [heightOfSpeedSpaceChart, setHeightOfSpeedSpaceChart] = useState(250);
-//   const [heightOfSimulationMap] = useState(MAP_MIN_HEIGHT);
-//   const [heightOfSpaceCurvesSlopesChart, setHeightOfSpaceCurvesSlopesChart] = useState(150);
-//   const [initialHeightOfSpaceCurvesSlopesChart, setInitialHeightOfSpaceCurvesSlopesChart] =
-//     useState(heightOfSpaceCurvesSlopesChart);
+  const [timeScaleDomain, setTimeScaleDomain] = useState<TimeScaleDomain>({
+    range: undefined,
+    source: undefined,
+  });
 
-//   const [timeScaleDomain, setTimeScaleDomain] = useState<TimeScaleDomain>({
-//     range: undefined,
-//     source: undefined,
-//   });
+  // X scale domain shared between SpeedSpace and SpaceCurvesSlopes charts.
+  const [positionScaleDomain, setPositionScaleDomain] = useState<PositionScaleDomain>({
+    initial: [],
+    current: [],
+    source: undefined,
+  });
 
-//   // X scale domain shared between SpeedSpace and SpaceCurvesSlopes charts.
-//   const [positionScaleDomain, setPositionScaleDomain] = useState<PositionScaleDomain>({
-//     initial: [],
-//     current: [],
-//     source: undefined,
-//   });
+  const {
+    selectedTrain,
+    selectedTrainRollingStock,
+    selectedTrainPowerRestrictions,
+    trainSimulation,
+    pathProperties,
+  } = useSimulationResults();
 
-//   const {
-//     allowancesSettings,
-//     selectedTrain,
-//     selectedProjection,
-//     simulation,
-//     simulationIsPlaying,
-//     dispatchUpdateSelectedTrainId,
-//     dispatchPersistentUpdateSimulation,
-//   } = useStoreDataForSpaceTimeChart();
+  const handleKey = (e: KeyboardEvent) => {
+    if (e.key === 'z' && e.metaKey) {
+      dispatch(persistentUndoSimulation());
+    }
+    if (e.key === 'e' && e.metaKey) {
+      dispatch(persistentRedoSimulation());
+    }
+  };
 
-//   const { data: selectedTrainSchedule } = osrdEditoastApi.endpoints.getTrainScheduleById.useQuery(
-//     {
-//       id: selectedTrain?.id as number,
-//     },
-//     { skip: !selectedTrain }
-//   );
+  useEffect(() => {
+    // Setup the listener to undi /redo
+    window.addEventListener('keydown', handleKey);
+    return function cleanup() {
+      window.removeEventListener('keydown', handleKey);
+      dispatch(updateTrainIdUsedForProjection(undefined));
+    };
```

Comment:
Remove this since we don't use the history of simulation anymore
```suggestion
```

Reply:
Done